src/client/features/search/index.js

```diff
@@ -95,22 +96,42 @@ class Search extends React.Component {
     this.getSearchResult();
   }
 
+  setAndSubmitSearchQuery(searchTerm) {
```

Comment:
Why is this function needed?

Reply:
It's used for the suggestions underneath the bar (e.g. TP53, Glycolysis, Ethanol). All those suggestions are clickable so I needed a way to set the current search query and submit it in one function.

Reply:
You mean like this function?

https://github.com/PathwayCommons/app-ui/pull/197/files#diff-a1f9f779cabf8780553a2f0a90378e43R76

Reply:
Yep, like that one. Didn't see it before. Just fixed it.

Reply:
You can just use:

```js
this.setQueryState({
  q: exampleText
});
```
It calls submitSearchQuery.

However, it isnt clear that ```setQueryState``` calls ```submitSearchQuery```, so you should keep the ```setQueryState``` method body, change the name to ```setAndSubmitSearchQuery``` and change every instance that ```setQueryState``` is used to ```setAndSubmitSearchQuery```.

The reason to use setQueryState instead of your version of the function is that it is more general.  E.g if you want to set query type or any other query field that function will handle that too.

Reply:
That makes sense. Better to keep things reusable. I've made the changes and committed them.